README.md

```diff
@@ -2,5 +2,248 @@
 
 [![Travis Build Status][build-badge]][build]
 
+This is a library of some custom validators for React components properties.
+Initially they were part of [React-Bootstrap](https://github.com/react-bootstrap/react-bootstrap/) project.
+
+### Usage
+
+All validators can be imported as
+```js
+import elementType from 'react-prop-types/elementType';
+// or
+import { elementType } from 'react-prop-types';
+...
+propTypes: {
+  someProp: elementType
+```
+or
+```js
+import CustomPropTypes from 'react-prop-types';
+// and then used as usual
+propTypes: {
+  someProp: CustomPropTypes.elementType
+```
+
+If you use `webpack`, you can bundle only those validators you need like this:
+```js
+import elementType from 'react-prop-types/elementType'
+```
+
+---
+#### all(arrayOfValidators)
+
+This validator allows to have complex validation like this:
+```js
+propTypes: {
+  vertical:  React.PropTypes.bool,
+  /**
+   * Display block buttons, only useful when used with the "vertical" prop.
+   * @type {bool}
+   */
+  block: CustomPropTypes.all([
+    React.PropTypes.bool,
+    function(props, propName, componentName) {
+      if (props.block && !props.vertical) {
+        return new Error('The block property requires the vertical property to be set to have any effect');
+      }
+    }
+  ])
+```
+
+All validators provided in an array as argument into `all()` will be validating
```

Comment:
Perhaps: All validators will be validated one by one, stopping on the first failure.